src/vmm/src/persist.rs

```diff
@@ -174,73 +173,47 @@ pub enum MicrovmStateError {
 }
 
 /// Errors associated with creating a snapshot.
-#[derive(Debug)]
+#[derive(Debug, thiserror::Error)]
 pub enum CreateSnapshotError {
     /// Failed to get dirty bitmap.
+    #[error("Cannot get dirty bitmap: {0}")]
     DirtyBitmap(VmmError),
     /// The virtio devices uses a features that is incompatible with older versions of Firecracker.
+    #[error(
+        "The virtio devices use a features that is incompatible with older versions of \
+         Firecracker: {0}"
+    )]
     IncompatibleVirtioFeature(&'static str),
     /// Invalid microVM version format
+    #[error("Invalid microVM version format")]
     InvalidVersionFormat,
     /// MicroVM version does not support snapshot.
+    #[error("Cannot translate microVM version to snapshot data version")]
     UnsupportedVersion,
     /// Failed to write memory to snapshot.
+    #[error("Cannot write memory file: {0}")]
     Memory(memory_snapshot::Error),
     /// Failed to open memory backing file.
+    #[error("Cannot perform {0} on the memory backing file: {1}")]
     MemoryBackingFile(&'static str, io::Error),
     /// Failed to save MicrovmState.
+    #[error("Cannot save the microVM state: {0}")]
     MicrovmState(MicrovmStateError),
     /// Failed to serialize microVM state.
+    #[error("Cannot serialize the microVM state: {0}")]
     SerializeMicrovmState(snapshot::Error),
     /// Failed to open the snapshot backing file.
+    #[error("Cannot perform {0} on the snapshot backing file: {1}")]
     SnapshotBackingFile(&'static str, io::Error),
-    #[cfg(target_arch = "x86_64")]
     /// Number of devices exceeds the maximum supported devices for the snapshot data version.
+    #[cfg(target_arch = "x86_64")]
+    #[error(
+        "Too many devices attached: {0}. The maximum number allowed for the snapshot data version \
+         requested is {FC_V0_23_MAX_DEVICES}."
+    )]
```

Comment:
This seems unrelated to the MAC change. We have a PR best practice to have one logical change per commit. Can this be split to a separate commit?